### DISCUSSION TITLE
Native task/function scheduling

### DISCUSSION THREAD

**Discussion by Harshal6927** [litestar-org/litestar#927] [Ideas]
Recently I had a requirement in my Django project were I had to run a function every 24 hours (specifically at 1 AM), I used Celery but their docs page was down; it was really frustrating to make it work. So I am suggesting adding this feature to Starlite.

Some key points:

- It should not effect main thread, however the task/function will be executed by main thread.

My proposed example:
```py
@shedule(invoke=2, at_utc=["09:12:00"], offset="+5:30")
def foo():
    ...
```

`invoke: int = 1`
`at_utc: list[str]`
`offset: str = None`

- This code snippet will run `2` times at `09:12:00`.
- Offset is optional  argument which can be use to change the default UTC time zone.
- When there is multiple time stings in `at_utc` list ie `at_utc=["09:12:00", "11:02:42"]` the function will be called/invoked at those two different time, in other words, it will run total of `4` times (2 times at 09:12:00 and 2 times at 11:02:42)

**Comment by Harshal6927** [comment]
The format of time string in `at_utc` will be in `HH:MM:SS`

**Comment by provinzkraut** [comment]
I don't think this should be a functionality native to Starlite. This is a very complex thing, we'd just be re-inventing the wheel and there's really not much that can be integrated with Starlite. The example you posted for instance is just an alternative way of describing a task. There's other scheduling libraries out there aside from Celery, and a library like that is what you really want to use.

The only thing that could be useful I think would be an integration for development environments, so you don't have to start worker processes manually.

**Reply by Harshal6927** [reply]
Got it
